src/OpenLoco/src/Vehicles/VehicleBogie.cpp

```diff
@@ -119,6 +104,164 @@ namespace OpenLoco::Vehicles
+
+                // Calculate new position - but don't update yet!! This is pushed to the stack.
+                World::Pos3 newPosition{ this->position.x + x_distance_hi, this->position.y + y_distance_hi, this->position.z - zDistance };
+
+                if (!sub_4AA959(this->position))
+                {
+                    World::Pos3 position_to_test = { newPosition.x, newPosition.y, this->position.z };
+                    if (sub_4AA959(position_to_test))
+                    {
+                        newPosition.x = this->position.x;
+                        newPosition.y = this->position.y;
+
+                        if (this->var_5A >= 0x50000)
+                        {
+                            rotateAndExplodeIfNotAlreadyExploded();
+                        }
+
+                        this->var_5A = ((this->var_5A & 0x7FFFFFFF) / 2) | 0x80000000;
+                    }
+
+                    if (sub_4AA959(newPosition))
+                    {
+                        newPosition.z = this->position.z;
+                        if (this->tileBaseZ >= 0x0A)
+                        {
+                            rotateAndExplodeIfNotAlreadyExploded();
+                        }
+
+                        this->tileBaseZ = 0;
+                    }
+                }
+
+                World::Pos2 currentPosition2D{ newPosition.x, newPosition.y };
+                auto newTileHeight = World::TileManager::getHeight(currentPosition2D);
+                if (newTileHeight.landHeight >= this->position.z
+                    || newTileHeight.landHeight >= newPosition.z)
+                {
+                    if (newTileHeight.landHeight < this->position.z
+                        && newTileHeight.landHeight >= newPosition.z)
+                    {
+                        newTileHeight.landHeight = this->position.z;
+                        newPosition.z = newTileHeight.landHeight;
+                        this->tileBaseZ = 0;
+                    }
+
+                    newPosition.x = this->position.x;
+                    newPosition.y = this->position.y;
+
+                    if (this->var_5A >= 0x50000)
+                    {
+                        rotateAndExplodeIfNotAlreadyExploded();
+                    }
+
+                    this->var_5A = ((this->var_5A & 0x7FFFFFFF) / 2) | 0x80000000;
+                }
+
+                if (newTileHeight.waterHeight != 0
+                    && newTileHeight.waterHeight < this->position.z
+                    && newTileHeight.waterHeight >= newPosition.z)
+                {
+                    if (rotateAndExplodeIfNotAlreadyExploded())
+                    {
+                        World::Pos3 splashPos{ this->position.x, this->position.y, newTileHeight.waterHeight };
+                        Splash::create(splashPos);
+                        Audio::playSound(Audio::SoundId::splash2, splashPos, 0x8001); // TODO: use kPlayAtLocation instead of hex literal for 0x8001
```

Comment:
you don't need the 0x8001 at all playSound has an overload that will apply all of this when you specify a position.